src/app/services/training/service.py

```diff
@@ -84,14 +69,15 @@ def build_pipeline(
             memory=memory,
         )
 
-    def train(self, file: UploadFile) -> MLModel:
-        X, y = run_pipeline(
-            file=file,
-            filter_config=self.filter_config,
-            mapping_config=self.mapping_config,
-            feature_engineer_config=self.feature_engineer_config,
-            selection_config=self.selection_config,
-        )
+    def train(self, training_file_path: Path) -> MLModel:
+
+        self.file_path = training_file_path
```

Comment:
Use an instance attribute only if the path must be reused across methods or exposed to callers, otherwise keep it local. If you need it on the model, declare it up front so Pydantic tracks it:
```suggestion
        // ...existing code...
        class TrainingService(BaseModel):
            model_path: Path = Field(default=Settings.MODEL_PATH)
            training_file_path: Path | None = None
        // ...existing code...
            def train(self, training_file_path: Path) -> MLModel:
                self.training_file_path = training_file_path
        // ...existing code...
```

If no consumer reads the value later, prefer 
```suggestion
        file_path = training_file_path
```
and avoid mutating the instance.